packages/engine/src/simulation/comms/package.rs

```diff
@@ -33,8 +34,12 @@ impl PackageComms {
         task: T,
         shared_store: TaskSharedStore,
     ) -> Result<ActiveTask> {
+        let task = task.into();
+        let task_name = task.get_task_name();
+
         self.inner
             .new_task(self.package_id, task, shared_store)
+            .instrument(tracing::trace_span!("Task", name = task_name))
```

Comment:
As per the PR description (this is generally about most of the spans, not just this one)
> Do we want to change the level of the spans? The EnvFilter can stop them being added if we're not in TRACE for example, same with texray not producing results if we're not in TRACE.

Reply:
I'd default to only add spans if `TRACE` is enabled. However, if we decide to guard spans behind a feature flag, the spans should always be enabled.

Reply:
My thoughts along this is that the spans aren't just for `texray` or timings and they aren't just for intrusive debugging (for instance with RUST_LOG=trace). Spans add a lot of contextual information that was missing from the logs, that we've been complaining about for a long time. 

For instance, when we move to hosting this in the cloud (or if other people choose to implement a different orchestration service), the spans provide good insights into the logging in production infrastructure. And they especially shine when ingesting in a logging service like DataDog or Splunk or similar.